src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/UpdateVCFSequenceDictionary.java

```diff
@@ -0,0 +1,150 @@
+                "(optional if the sequence dictionary source is specified as a reference argument). The dictionary " +
+                "presented must be valid (contain a sequence record) for each sequence that is referenced by any " +
+                "variant in the input file.",
+            optional=true)
+    String dictionarySource; // optional since a reference can be provided as a dictionary source instead
+
+    public final static String replaceArgumentName = "replace";
+    @Argument(fullName=replaceArgumentName,
+            shortName=replaceArgumentName,
+            doc="Force replacement of the dictionary if the input file already has a dictionary",
+            optional=true)
+    boolean replace = false; // optional since a reference can be provided as a dictionary source instead
+
+    private VariantContextWriter vcfWriter;
+    private SAMSequenceDictionary sourceDictionary;
+
+    @Override
+    public void onTraversalStart() {
+        VCFHeader vcfHeader = getHeaderForVariants();
+        sourceDictionary = getSequenceDictionaryFromInput(dictionarySource);
+
+        // Warn and require opt-in via -replace if we're about to clobber a valid sequence
+        // dictionary. Check the input file directly via the header rather than using the
+        // engine, since it might dig one up from an index.
+        SAMSequenceDictionary oldDictionary =
+                vcfHeader == null ? null : vcfHeader.getSequenceDictionary();
+        if ( (oldDictionary != null && !oldDictionary.getSequences().isEmpty()) && !replace) {
+            throw new UserException.BadArgumentValue(
+                    String.format(
+                            "The input variant file %s already contains a sequence dictionary. " +
+                            "Use %s to force the dictionary to be replaced.",
+                            getDrivingVariantsFeatureInput().getName(),
+                            replaceArgumentName
+                    )
+            );
+        }
+
+        vcfHeader.setSequenceDictionary(sourceDictionary);
+
+        vcfWriter = createVCFWriter(new File(outFile));
+        vcfWriter.writeHeader(vcfHeader);
+    }
+
+    @Override
+    public void apply(final VariantContext vc, final ReadsContext readsContext, final ReferenceContext ref, final FeatureContext featureContext) {
+        // Validate each variant against the source dictionary manually
+        SAMSequenceRecord samSeqRec = sourceDictionary.getSequence(vc.getContig());
+        if (samSeqRec == null) {
+            throw new UserException.BadArgumentValue(
+                String.format(
+                    "The input variant file contains a variant (ID: \"%s\") with a reference to a sequence (\"%s\") " +
+                    "that is not present in the provided dictionary",
+                    vc.getID(),
+                    vc.getContig()
+                )
+            );
+        } else if (vc.getEnd() > samSeqRec.getSequenceLength()) {
```

Comment:
So on further thought, I don't think it's actually valid at all to have a structural variant off the end.  Maybe leave this as is and if someone shows up with a file that has them we can reconsider.  I was looking at the spec to try to tell for sure, but it seems like end should always be on the reference.  Sorry for the confusion.